### PR TITLE
CLZ and CTZ for non power of two inputs

### PR DESCRIPTION
**Pull request by Hazardu**
In this PR i add zeros to input signals of `count_leading_zeros` to extend the shape to nearest power of two

Original issue:
https://github.com/kuznia-rdzeni/coreblocks/issues/751

### CODE REVIEW

File: transactron/utils/amaranth_ext/functions.py [hunk 1]
```diff
@@ -53,28 +53,24 @@ def iter(s: Value, step: int) -> Value:
...
-
-    value = iter(s, xlen_log)
+    slen = len(s)
+    slen_log = ceil_log2(slen)
+    closest_pow_2_of_s = 2 ** slen_log
+    zeros_prepend_count = closest_pow_2_of_s - slen
+    if zeros_prepend_count > 0:
+        value = iter(Cat(C(0, shape=zeros_prepend_count), s), slen_log) 
+    else:
+        value = iter(s, slen_log)
```
Comment (tilk) [inline transactron/utils/amaranth_ext/functions.py:63]: I think the `if` is unnecessary and the `else` case can be removed.

**Review by tilk — COMMENTED** [review]
There is a merge conflict.

File: test/utils/test_utils.py [hunk 2]
```diff
@@ -109,7 +110,7 @@ def elaborate(self, platform):
         return m
 
 
-@pytest.mark.parametrize("size", range(1, 7))
+@pytest.mark.parametrize("size", range(1, 128))
```
Comment (tilk) [inline test/utils/test_utils.py:113]: Reduce the number of tests.

File: test/utils/test_utils.py [hunk 3]
```diff
@@ -152,7 +154,7 @@ def elaborate(self, platform):
...
```
Comment (Arusekk) [inline test/utils/test_utils.py:157]: Notes from discussion:

```suggestion
@pytest.mark.parametrize("size", [1,2,3,4,5,6,7,8,9,63,64,65,97,98,127,128])
```
(less tests sample)

**Review by Arusekk — APPROVED** [review]

**Pull request merged**